hazelcast/src/main/java/com/hazelcast/cache/impl/wan/WanCacheEntryView.java

```diff
@@ -71,25 +73,31 @@ public WanCacheEntryView(@Nonnull Data dataKey,
     @Override
     public K getKey() {
         if (key == null) {
-            key = serializationService.toObject(dataKey);
+            key = serializationService.toObject(getDataKey());
```

Comment:
Do we really need to trim the schema before deserailising here? Same for value below and same in `WanMapEntryView`.

Reply:
If we don't trim the schema, Data's are not equal for the same object anymore. We rely on the hashcode of the Data for the keys. And also there are places where we check equality over byte-array even for value. 

I have attempted to change Data so that it will skip schema part, but that get complicated pretty quickly. I have decided not to proceed with that.

Reply:
Ok, I guess that's understandable, I just thought that if you deserialise an object that contains the schema and if you deserialise an object that doesn't contain the schema, that you end up with the same object. So the question was actually - do we need to trim the schema before deserialising it and returning it to the user?

Reply:
> So the question was actually - do we need to trim the schema before deserialising it and returning it to the user?
I think I understand now. We don't need to create the object from `trimmedDataKey`. `dataKey` suffice. I will change this one.

Reply:
Great, thanks!

Reply:
Fixed by 39195dc